src/server.js

```diff
@@ -170,103 +174,195 @@ api.post('/getChunkedData', (req, res) => {
   }
   console.log(`dataPath = ${dataPath}`);
 
-  // call 'vg chunk' to generate graph
-  let vgChunkParams = ['chunk', '-x', `${dataPath}${xgFile}`];
-  if (req.withGam) {
-    // Use a GAM index
-    vgChunkParams.push('-a', `${dataPath}${gamFile}`, '-g');
+  // parse region
+  // either a range -> seq:start-end
+  // or a position and distance -> seq:post+distance
+  let region = req.body.region;
+  let region_col = region.split(":");
+  let start_end = region_col[1].split("-");
+  let pos_dist = region_col[1].split("+");
+  let r_start = -1;
+  let r_end = -1;
+  let distance = -1;
+  if(start_end.length == 2){
+    r_start = Number(start_end[0]);
+    r_end = Number(start_end[1]);
+  } else if(pos_dist.length == 2){
+    r_start = Number(pos_dist[0]);
+    distance = Number(pos_dist[1]);
   }
-  if (req.withGbwt) {
-    // Use a GBWT haplotype database
-    vgChunkParams.push('--gbwt-name', `${dataPath}${gbwtFile}`);
+
+  // check the bed file if this region has been pre-fetched
+  let chunkPath = '';
+  if(req.withBed){
+    let i = 0;
+    let nb_regions_in_bed = 0;
+    if(req.body.regionInfo && Object.keys(req.body.regionInfo).length > 0){
+      nb_regions_in_bed = req.body.regionInfo['desc'].length;
+    }
+    while (i < nb_regions_in_bed && chunkPath === ''){
+      let region_chr = req.body.regionInfo['chr'][i];
+      let region_start = req.body.regionInfo['start'][i];
+      let region_end = req.body.regionInfo['end'][i];
+      if(region_chr.concat(':', region_start, '-', region_end) === region){
+	if(req.body.regionInfo['chunk'][i] !== ''){
+	  chunkPath = req.body.regionInfo['chunk'][i]
+	}
+      }
+      i += 1;
+    }
+    // check that the 'chunk.vg' file exists in the chunk folder
+    let chunk_file = `${dataPath}${chunkPath}/chunk.vg`;
+    if(fs.existsSync(chunk_file)){
+      console.log(`found pre-fetched chunk at ${chunk_file}`);
+    } else {
+      console.log(`couldn't find pre-fetched chunk at ${chunk_file}`);
+      chunkPath = '';
+    }
   }
-  const position = Number(req.body.nodeID);
-  const distance = Number(req.body.distance);
-  if (
-    Object.prototype.hasOwnProperty.call(req.body, 'byNode') &&
-    req.body.byNode === 'true'
-  ) {
-    vgChunkParams.push('-r', position, '-c', distance);
-  } else {
+  
+  if(chunkPath === ''){
+    // call 'vg chunk' to generate graph
+    let vgChunkParams = ['chunk', '-x', `${dataPath}${xgFile}`];
```

Comment:
We might be missing some path sanitization here too.

Reply:
https://github.com/vgteam/sequenceTubeMap/pull/140/commits/8db286b9f20863a16979c82155aa6532bd04a9ae adds sanitizing of the paths by checking that the path is allowed and the files have the expected extension. Here that `xgFile` ends with `.xg`